### PR TITLE
debuff effect & 19FAF0 funcs

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 13 functions (+0.15%) on `us`.

✅ This PR fixes 5 warnings!

File: src/19FAF0.c [hunk 1]
```diff
@@ -13,25 +13,69 @@ ApiStatus func_80271258(Evt* script, s32 isInitialCall) {
...
+
+ApiStatus func_80271588(Evt* script, s32 isInitialCall) {
+    s32 i;
+
+    for (i = 0; i < 20; i++) {
+        fx_floating_cloud_puff(0,
+                               script->varTable[0] + rand_int(30) - 15,
+                               script->varTable[1] + rand_int(20) - 15,
+                               script->varTable[2] + 5,
+                               1.0f, 25);
```
Comment (nanaian) [inline src/19FAF0.c:75]:
```suggestion
                               1.0f,
                               25
        );
```

File: src/19FAF0.c [hunk 2]
```diff
@@ -126,12 +172,130 @@ void dispatch_event_player_continue_turn(s32 eventType) {
...
+    ActorState* state = &player->state;
+    Actor* target;
+    ActorPart* part;
+
+    battleStatus->currentTargetID2 = battleStatus->currentTargetID;
+    battleStatus->currentTargetPart2 = battleStatus->currentTargetPart;
+
+    target = get_actor(currentTargetID);
+
+
```
Comment (nanaian) [inline src/19FAF0.c:189]:
```suggestion
```

File: src/19FAF0.c [hunk 3]
```diff
@@ -126,12 +172,130 @@ void dispatch_event_player_continue_turn(s32 eventType) {
...
+    ActorState* state = &player->state;
+    s32 oldHPChangeCounter;
+    s32 flags;
+    s32 dispatchEvent;
+    s32 oldPlayerHP;
+    s32 temp;
+
+    battleStatus->currentAttackDamage = damageAmount;
+
+    temp = (s16)damageAmount; // usage of temp here required to match
```
Comment (nanaian) [inline src/19FAF0.c:242]: We're sure dispatch_damage_event_player has the right signature?

File: src/19FAF0.c [hunk 4]
```diff
@@ -220,11 +463,213 @@ ApiStatus GetPlayerHP(Evt* script, s32 isInitialCall) {
...
+    if (flags & BS_FLAGS1_80) {
+        gBattleStatus.flags1 |= BS_FLAGS1_80;
+    } else {
+        gBattleStatus.flags1 &= ~BS_FLAGS1_80;
+    }
+    if (flags & BS_FLAGS1_800) {
+        gBattleStatus.flags1 |= BS_FLAGS1_800;
+    } else {
+        gBattleStatus.flags1 &= ~BS_FLAGS1_800;
+    }
```
Comment (nanaian) [inline src/19FAF0.c:655]: This is a great case for us allowing brace-less ifs I think; this form appears a lot in other places too. or we could macro it like `SET_FLAG(flags & BS_FLAGS_800, gBattleStatus.flags1, BS_FLAGS1_800);`

**Review by nanaian — CHANGES_REQUESTED** [review]

File: src/19FAF0.c (hunk 3 above)
Comment (ethteck) [reply src/19FAF0.c:242]: yeah, damageAmount can't be s16 cause it messes up callers

**Pull request merged**